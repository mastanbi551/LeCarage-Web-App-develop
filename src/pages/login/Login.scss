@import "../../styles/variables.scss";
.login-page {
  &__heading {
    padding: 20px 20px 20px 0;
  }

  &__container {
    display: flex;
    flex-direction: row;
    width: 100%;
  }

  &__logo-container {
    flex-basis: 20%;
    padding-top: 3%;
    width: 100%;
  }

  &__logo-img {
    /*to be updated.*/
    color: transparent;
    width: 100%;
  }

  &__login-form {
    padding: 50px;
    margin-left: 10%;

    &__user-name-field {
      padding: 0 20px 20px 0;
    }

    &__user-password-field {
      padding: 0 20px 20px 0;
    }

    &__checkbox-forgotpassword {
      padding: 0 0 20px 0;
      display: flex;
      width: 318px;
      justify-content: space-between;
      @include checkbox-forgotpassword;
    }

    &__forgot-password {
      &__content {
        @include Body_Copy_Main;
        color: $darkgrey;
        cursor: pointer;
      }
    }

    &__form-fields {
      display: flex;
      flex-direction: column;
      align-items: flex-start;
      width: 100%;
    }

    &__submit-button {
      margin-top: 6%;
    }
  }

  &__checkbox-forgot {
    display: flex;
  }

  .checkbox-btn {
    color: #f1f1f1;
  }
  .checkbox-text {
    padding-left: 0px;
  }
  .checkbox-btn {
    color: $primaryWhite;
    font-weight: 400;
    font-size: 12px;
    line-height: 18px;
    padding-top: 3px;
    display: flex;
    align-items: center;
  }

  .form-checkbox {
    margin-right: 10px;
    vertical-align: baseline;
  }

  .checkbox-text {
    padding-left: 10px;
  }

  input[type="checkbox"] {
    background-color: rgba(255, 255, 255, 0);
    appearance: none;
    border: 1px solid $primaryWhite;
    color: $primaryBlack;
    width: 15px;
    height: 15px;
    border-radius: 2px;
  }

  input[type="checkbox"]:checked {
    background: $primaryGreen; //rgba(36, 39, 209, 0.7);
    color: $primaryBlack;
    border: 1px solid $primaryGreen;
    border-radius: 2px;
    border-radius: 2px;
  }

  input[type="checkbox"]:checked::before {
    content: "\2713";
    font-size: 13px;
    font-weight: 800;
    color: $primaryWhite;
    position: relative;
    top: -2px;
    left: 2px;
  }
  .static {
    border: 1px solid $lightgrey;
    background-color: transparent;
    box-sizing: border-box;
    width: 100%;
    height: 40px;
    min-width: 19.875rem;
  }
}
.input-text-field span.loginicon i:last-child svg {
  color: $primaryWhite !important;
}

