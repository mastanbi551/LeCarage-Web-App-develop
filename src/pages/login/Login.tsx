import React from 'react';
import { Formik } from 'formik';
import {
  BackgroundImageContainer,
  InputTextField,
  PageHeader,
  LeCarageLogo,
  CustomButton,
  CustomCheckBox,
  Loader,
} from '../../components';
import { ImagesPath, ConstantStrings } from '../../utils/constants';
import {
  AiOutlineEyeInvisible,
} from 'react-icons/ai';
import './Login.scss';
import { LoginSchema } from '../../utils/ValidationSchemas';
import { loginValidation } from '../../utils';
import { useNavigate } from 'react-router';
import { HiOutlineUser } from 'react-icons/hi';
import { BiLockAlt } from 'react-icons/bi';
import { HiEye } from 'react-icons/hi';
import _ from 'lodash';
interface IProps {
  passwordType: string;
  handleOnSubmit: (values: {
    email: string;
    password: string;
    isRemember: boolean;
  }) => void;
  handleInputChange: () => void;
  userData?: {
    email: string;
    password: string;
    isRemember: boolean;
  };
  showLoader: boolean
}

export const Login: React.FC<IProps> = ({
  passwordType,
  handleOnSubmit,
  handleInputChange,
  userData,
  showLoader
}) => {
  const navigate = useNavigate();
  return (
    <BackgroundImageContainer imageSource={ImagesPath.bgImageWithCar}>
      <Formik
        initialValues={{
          email: userData ? userData.email : '',
          password: userData ? userData.password : '',
          isRemember: userData ? userData.isRemember : false
        }}
        validationSchema={LoginSchema}
        onSubmit={(values) => {
          handleOnSubmit(values);
        }}
      >
        {({
          handleChange,
          handleSubmit,
          values,
          errors,
          isValid,
          setFieldValue,
          handleBlur,
          touched,

        }) => (
          <div className='login-page col-md-12'>
            <div className='login-page__container container-fluid'>
              <div className='login-page__login-form  col-md-6'>
                <div className='login-page__heading' data-testid='pageheader'>
                  <PageHeader
                    text={ConstantStrings.loginPageHeading}
                    name='heading'
                  />
                </div>
                <form
                  onSubmit={handleSubmit}
                  onKeyDown={(e) => {
                    if (e.key === 'Enter')
                      handleSubmit();
                  }}
                  className='col-md-12 login-page__login-form__form-fields' data-testid='login-form'
                >
                  <div className='login-page__login-form__user-name-field '>
                    <InputTextField
                      dataTestId='input-field'
                      name='input-field-icon'
                      id='email_field'
                      placeholder={ConstantStrings.emailPlaceholder}
                      value={values.email}
                      inputType='text'
                      requiredOrNot={true}
                      onChangeText={handleChange('email')}
                      onBlur={handleBlur('email')}
                      error={errors.email}
                      touched={touched.email}
                    >
                      <span className='single-icon'>
                        <i>
                          <HiOutlineUser />
                        </i>
                      </span>
                    </InputTextField>

                  </div>

                  <div className="  login-page__login-form__user-password-field ">
                    <InputTextField
                      dataTestId='password'
                      name='input-field-icon'
                      id='password_field'
                      placeholder={ConstantStrings.passwordPlaceholder}
                      value={values.password}
                      inputType={passwordType}
                      requiredOrNot={true}
                      onChangeText={handleChange('password')}
                      onBlur={handleBlur('password')}
                      error={errors.password}
                      touched={touched.password}

                    >
                      <span className='multiple-icons loginicon'>
                        <i><BiLockAlt /></i>
                        <i onClick={() => {
                          handleInputChange();
                        }}>
                          {passwordType ? <HiEye /> : <AiOutlineEyeInvisible />}
                        </i>
                      </span>

                    </InputTextField>

                  </div>
                  <div className='login-page__login-form__checkbox-forgotpassword'>
                    <div className='login-page__login-form__rememberme-checkbox col-md-6'>
                      <CustomCheckBox
                        name='form-checkbox col-md-12'
                        value={_.toString(values.isRemember)}
                        onClick={() => setFieldValue('isRemember', !values.isRemember)} 
                        checkBoxText='Remember Me'
                        checkedOrNot={values.isRemember}
                      />
                    </div>
                    <div className='login-page__login-form__forgot-password col-md-6'>
                      <span className='login-page__login-form__forgot-password__content' onClick={(e) => {
                        e.stopPropagation();
                        navigate('/forgotpassword');
                      }}> Forgot Password?</span>
                    </div>
                  </div>
                  <div className='login-page__login-form__submit-button' >
                    <CustomButton
                      data-testid='buttonsub'
                      title="LOG IN"
                      name={showLoader ? 'cursor-not-allowed button_full': 'button_full'} 
                      buttonOnClick={handleSubmit}
                      isDisable={!isValid || !loginValidation(values.email, values.password)}
                    />
                  </div>
                </form>
              </div>
              {
                showLoader && <Loader />
              }
              <div className='login-page__logo-container col-md-6'>
                <LeCarageLogo
                  source={ImagesPath.lecaragelogo}
                  altText='logo'
                  name='logo-page__img'
                />
              </div>
            </div>
          </div>
        )}
      </Formik>
    </BackgroundImageContainer>
  );

};
