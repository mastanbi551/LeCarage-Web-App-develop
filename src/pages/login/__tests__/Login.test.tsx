import React from 'react';
import { render, fireEvent, getByTestId  } from '@testing-library/react';


import { Login } from '../Login';
import { BrowserRouter, Router } from 'react-router-dom';

describe('Login component', () => {

    it('renders the email and password input fields', () => {
        const setAuth = jest.fn();
        const passwordType = 'text';
        const handleOnSubmit = jest.fn();
        const handleInputChange = jest.fn();
        const handleCheckBoxClick = jest.fn();
    
        const { getByTestId } = render(
            <BrowserRouter><Login
            //setAuth={setAuth}
            passwordType={passwordType}
            handleOnSubmit={handleOnSubmit}
            handleInputChange={handleInputChange}
            //handleCheckBoxClick={handleCheckBoxClick}
          /></BrowserRouter>
        );
    
        const emailInput = getByTestId('input-field');
        const passwordInput = getByTestId('password');
    
        expect(emailInput).toBeInTheDocument();
        expect(passwordInput).toBeInTheDocument();
      });
      it('updates the values of email and password when input values are changed', () => {
        const setAuth = jest.fn();
        const passwordType = 'text';
        const handleOnSubmit = jest.fn();
        const handleInputChange = jest.fn();
        const handleCheckBoxClick = jest.fn();
    
        const { getByTestId } = render(
            <BrowserRouter><Login
           // setAuth={setAuth}
            passwordType={passwordType}
            handleOnSubmit={handleOnSubmit}
            handleInputChange={handleInputChange}
           // handleCheckBoxClick={handleCheckBoxClick}
          /></BrowserRouter>
        );
    
        const emailInput = getByTestId('input-field') as HTMLFormElement;
        const passwordInput = getByTestId('password') as HTMLFormElement;
    
        fireEvent.change(emailInput, { target: { value: 'test@email.com' } });
        fireEvent.change(passwordInput, { target: { value: 'password123' } });
    
        expect(emailInput.value).toBe('test@email.com');
        expect(passwordInput.value).toBe('password123');
      });
    


  it('matches snapshot', () => {
    const setAuth = jest.fn();
    const passwordType = 'password';
    const handleOnSubmit = jest.fn();
    const handleInputChange = jest.fn();
    const handleCheckBoxClick = jest.fn();
  
    // Render the component and take a snapshot of the output
    const { container } = render(
        <BrowserRouter><Login
        //setAuth={setAuth}
        passwordType={passwordType}
        handleOnSubmit={handleOnSubmit}
        handleInputChange={handleInputChange}
       // handleCheckBoxClick={handleCheckBoxClick}
      /></BrowserRouter>
    );

    // Compare the output with the saved snapshot
    expect(container).toMatchSnapshot();
  });
});
