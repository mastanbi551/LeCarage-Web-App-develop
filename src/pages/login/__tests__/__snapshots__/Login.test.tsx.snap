// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component matches snapshot 1`] = `
<div>
  <div
    data-testid="background-image-container"
    style="background-image: url(bgimg.jpg); background-size: cover; background-repeat: no-repeat; background-position: center; height: 100vh; overflow: hidden;"
  >
    <div
      class="login-page col-md-12"
    >
      <div
        class="login-page__container container-fluid"
      >
        <div
          class="login-page__login-form  col-md-6"
        >
          <div
            class="login-page__heading"
            data-testid="pageheader"
          >
            <h1
              class="heading"
            >
              Log In
            </h1>
          </div>
          <form
            class="col-md-12 login-page__login-form__form-fields"
            data-testid="login-form"
          >
            <div
              class="login-page__login-form__user-name-field "
            >
              <div
                class="input-text-field"
              >
                <input
                  autocomplete="new-password"
                  autocorrect="off"
                  class="input-field-icon static"
                  data-testid="input-field"
                  id="email_field"
                  placeholder="Enter Email"
                  required=""
                  type="text"
                  value=""
                />
                <span
                  class="single-icon"
                >
                  <i>
                    <svg
                      fill="none"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M16 7a4 4 0 11-8 0 4 4 0 018 0zM12 14a7 7 0 00-7 7h14a7 7 0 00-7-7z"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                    </svg>
                  </i>
                </span>
              </div>
            </div>
            <div
              class="  login-page__login-form__user-password-field "
            >
              <div
                class="input-text-field"
              >
                <input
                  autocomplete="new-password"
                  autocorrect="off"
                  class="input-field-icon static"
                  data-testid="password"
                  id="password_field"
                  placeholder="Enter Password"
                  required=""
                  type="password"
                  value=""
                />
                <span
                  class="multiple-icons loginicon"
                >
                  <i>
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12 2C9.243 2 7 4.243 7 7v3H6c-1.103 0-2 .897-2 2v8c0 1.103.897 2 2 2h12c1.103 0 2-.897 2-2v-8c0-1.103-.897-2-2-2h-1V7c0-2.757-2.243-5-5-5zm6 10 .002 8H6v-8h12zm-9-2V7c0-1.654 1.346-3 3-3s3 1.346 3 3v3H9z"
                      />
                    </svg>
                  </i>
                  <i>
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 20 20"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M10 12a2 2 0 100-4 2 2 0 000 4z"
                      />
                      <path
                        clip-rule="evenodd"
                        d="M.458 10C1.732 5.943 5.522 3 10 3s8.268 2.943 9.542 7c-1.274 4.057-5.064 7-9.542 7S1.732 14.057.458 10zM14 10a4 4 0 11-8 0 4 4 0 018 0z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </i>
                </span>
              </div>
            </div>
            <div
              class="login-page__login-form__checkbox-forgotpassword"
            >
              <div
                class="login-page__login-form__rememberme-checkbox col-md-6"
              >
                <div
                  class="checkbox-btn"
                >
                  <input
                    class="form-checkbox col-md-12"
                    data-testid="my-checkbox"
                    type="checkbox"
                    value="false"
                  />
                  <span
                    class="checkbox-text"
                    data-testid="my-div"
                  >
                    Remember Me
                  </span>
                </div>
              </div>
              <div
                class="login-page__login-form__forgot-password col-md-6"
              >
                <span
                  class="login-page__login-form__forgot-password__content"
                >
                   Forgot Password?
                </span>
              </div>
            </div>
            <div
              class="login-page__login-form__submit-button"
            >
              <button
                class="button_full"
                type="button"
              >
                LOG IN
              </button>
            </div>
          </form>
        </div>
        <div
          class="login-page__logo-container col-md-6"
        >
          <img
            alt="logo"
            class="logo-page__img"
            src="lecarageLogo.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
